### PR TITLE
congestion: fine-tuning parameters in the model

### PR DESCRIPTION
**Pull request by jakmeier**
This adds the ability to parametrize the NEP strategy.
Plus, it add sever model parametrizations for comparisons..

### CODE REVIEW

**Comment by wacban** [comment]
> But for the balanced workload with more than 4 shards, we lose a lot of utilization this way. (60% with 6 shards, 40% with 10 shards) 

This is surprising, I would expect under balanced load the utilization to be always at 100%. The pure symmetry of this workload should mean that all shards are equally congested and we always have work to do. 

By comparison I would expect a combination of AllToOne and Non-Congesting-Balanced to have lowered utilization because of global stop that prevents non congested shards from processing. 

btw composable workloads is an idea

Regarding the tradeoff my preference would be to keep just a few chunks worth of work in the buffers (e.g. 2 in incoming, 2 in outgoing) using some average attached / burned ratio (instead of pessimistic). This should give us reasonable bound on receipt latency (e.g. 4 blocks) and some weak guarantees on utilization under typical load.

File: tools/congestion-model/src/main.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+use bytesize::ByteSize;
```
Comment (wacban) [inline tools/congestion-model/src/main.rs:1]: ❤️

**Review by wacban — APPROVED** [review]
LGTM

**Comment by jakmeier** [comment]
> > But for the balanced workload with more than 4 shards, we lose a lot of utilization this way. (60% with 6 shards, 40% with 10 shards)
> 
> This is surprising, I would expect under balanced load the utilization to be always at 100%. The pure symmetry of this workload should mean that all shards are equally congested and we always have work to do.
> 
> By comparison I would expect a combination of AllToOne and Non-Congesting-Balanced to have lowered utilization because of global stop that prevents non congested shards from processing.

Yeah, it makes no sense, I had to dig deeper to find out what's happening. Turns out I was too aggressive with dropping unprocessed transactions from the pool. That can give cyclical behaviour where some portion of the rounds have not enough transactions to accept. Funny enough, more shards can sometime be better and sometime worse, depends on how the cycles interfere.


> btw composable workloads is an idea

We absolutely need to do it, yes. But I have been holding myself back, as I think we should do that level of details with Locus on real nearcore, rather than on the model. That's just my opinion, let me know id you think otherwise.

> Regarding the tradeoff my preference would be to keep just a few chunks worth of work in the buffers (e.g. 2 in incoming, 2 in outgoing) using some average attached / burned ratio (instead of pessimistic). This should give us reasonable bound on receipt latency (e.g. 4 blocks) and some weak guarantees on utilization under typical load.

I like that approach!

**Pull request merged**